File: eKucniLjubimci.Services/Database/Zivotinja.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace eKucniLjubimci.Services.Database
{
    public class Zivotinja
    {
        public int ZivotinjaId { get; set; }
        public string Naziv { get; set; }
        public string Napomena { get; set; }
        public decimal Cijena { get; set; }
        public bool Dostupnost { get; set; }
        public DateTime DatumPostavljanja { get; set; }
        public string StateMachine { get; set; }

        public int? NarudzbaId { get; set; }
        public virtual Narudzba Narudzba { get; set; }

        public int VrstaId { get; set; }
        public virtual Vrsta Vrsta { get; set; }

        public virtual List<Slika> Slike { get; set; } = new List<Slika>();

    }
}

File: flutterUI/flutter_desktop/lib/models/zivotinja.dart
import 'package:flutter_desktop/models/slika.dart';
import 'package:flutter_desktop/models/vrsta.dart';
import 'package:json_annotation/json_annotation.dart';

part 'zivotinja.g.dart';

@JsonSerializable()
class Zivotinja {
  int? zivotinjaId;
  String? naziv;
  String? napomena;
  double? cijena;
  bool? dostupnost;
  DateTime? datumPostavljanja;
  String? stateMachine;
  int? vrstaId;
  Vrsta? vrsta;
  List<Slika>? slike;

  Zivotinja({
    this.zivotinjaId,
    this.naziv,
    this.napomena,
    this.cijena,
    this.dostupnost,
    this.datumPostavljanja,
    this.stateMachine,
    this.vrstaId,
    this.vrsta,
    this.slike,
  });

  factory Zivotinja.fromJson(Map<String, dynamic> json) =>
      _$ZivotinjaFromJson(json);

  Map<String, dynamic> toJson() => _$ZivotinjaToJson(this);
}
